extensions/wikia/Recirculation/js/views/mixedFooter.js

```diff
@@ -115,37 +98,18 @@ define('ext.wikia.recirculation.views.mixedFooter', [
 		});
 	}
 
-	function getTemplateList(newsAndStoriesArticles) {
-		var templateList = [templatePaths.article, templatePaths.sponsoredContent],
-			topicExist = false,
-			storyStreamExist = false;
-
-		newsAndStoriesArticles.forEach(function (article) {
-			if (article.type === 'topic' && !topicExist) {
-				templateList.push(templatePaths.topic);
-				topicExist = true;
-			} else if (article.type === 'storyStream' && !storyStreamExist) {
-				templateList.push(templatePaths.storyStream);
-				storyStreamExist = true;
-			}
-		});
-
-		return templateList;
-	}
-
 	function setupTracking() {
-		track({
-			action: tracker.ACTIONS.IMPRESSION
-		});
+		tracker.trackImpression('footer');
 
 		$mixedContentFooter.on('click', '[data-tracking]', function () {
-			var labels = $(this).data('tracking').split(',');
+			var $this = $(this),
+				labels = $this.data('tracking').split(','),
+				href = $this.attr('href');
+
 			labels.forEach(function (label) {
-				track({
-					action: tracker.ACTIONS.CLICK,
-					label: label
-				});
+				tracker.trackClick(label);
 			});
+			tracker.trackSelect(href);
```

Comment:
would it be reasonable to extract that callback as we use the same for rail and mcf?